Std/Data/Nat/Lemmas.lean

```diff
@@ -133,85 +133,170 @@ theorem recDiagOn_succ_succ {motive : Nat → Nat → Sort _} (zero_zero : motiv
+theorem ne_of_lt' {a b : Nat} (h : b < a) : a ≠ b := Nat.ne_of_gt h
 
 @[simp] protected theorem not_le {a b : Nat} : ¬ a ≤ b ↔ b < a :=
   ⟨Nat.gt_of_not_le, Nat.not_le_of_gt⟩
 
 @[simp] protected theorem not_lt {a b : Nat} : ¬ a < b ↔ b ≤ a :=
   ⟨Nat.ge_of_not_lt, flip Nat.not_le_of_gt⟩
 
-theorem le_lt_antisymm {n m : Nat} (h₁ : n ≤ m) (h₂ : m < n) : False :=
-  Nat.lt_irrefl n (Nat.lt_of_le_of_lt h₁ h₂)
+protected alias ⟨lt_of_not_ge, _⟩ := Nat.not_le
+protected alias ⟨lt_of_not_le, _⟩ := Nat.not_le
+protected alias ⟨_, not_le_of_lt⟩ := Nat.not_le
+protected alias ⟨_, lt_le_antisymm⟩ := Nat.not_le
 
-theorem lt_le_antisymm {n m : Nat} (h₁ : n < m) (h₂ : m ≤ n) : False :=
-  le_lt_antisymm h₂ h₁
+protected alias ⟨le_of_not_gt, _⟩ := Nat.not_lt
+protected alias ⟨le_of_not_lt, _⟩ := Nat.not_lt
+protected alias ⟨_, not_lt_of_ge⟩ := Nat.not_lt
+protected alias ⟨_, not_lt_of_le⟩ := Nat.not_lt
+protected alias ⟨_, le_lt_antisymm⟩ := Nat.not_lt
 
-protected theorem lt_asymm {n m : Nat} (h₁ : n < m) : ¬ m < n :=
-  le_lt_antisymm (Nat.le_of_lt h₁)
+protected theorem le_of_not_ge {a b : Nat} (h : ¬ b ≤ a) : a ≤ b := Nat.le_of_lt (Nat.not_le.1 h)
+protected alias le_of_not_le := Nat.le_of_not_ge
 
-/-- Strong case analysis on `a < b ∨ b ≤ a` -/
-protected def lt_sum_ge (a b : Nat) : a < b ⊕' b ≤ a :=
-  if h : a < b then .inl h else .inr (Nat.not_lt.1 h)
+protected theorem lt_asymm {a b : Nat} (h : a < b) : ¬ b < a := Nat.not_lt.2 (Nat.le_of_lt h)
+protected alias not_lt_of_gt := Nat.lt_asymm
+protected alias not_lt_of_lt := Nat.lt_asymm
 
-/-- Strong case analysis on `a < b ∨ a = b ∨ b < a` -/
-protected def sum_trichotomy (a b : Nat) : a < b ⊕' a = b ⊕' b < a :=
-  match a.lt_sum_ge b with
-  | .inl h => .inl h
-  | .inr h₂ => match b.lt_sum_ge a with
-    | .inl h => .inr <| .inr h
-    | .inr h₁ => .inr <| .inl <| Nat.le_antisymm h₁ h₂
+protected theorem lt_iff_le_and_not_ge {a b : Nat} : a < b ↔ a ≤ b ∧ ¬ b ≤ a :=
+  ⟨fun h => ⟨Nat.le_of_lt h, Nat.not_le_of_gt h⟩, fun ⟨_, h⟩ => Nat.lt_of_not_ge h⟩
 
-protected theorem lt_trichotomy (a b : Nat) : a < b ∨ a = b ∨ b < a :=
-  match a.sum_trichotomy b with
-  | .inl h => .inl h
-  | .inr (.inl h) => .inr (.inl h)
-  | .inr (.inr h) => .inr (.inr h)
+protected alias lt_iff_le_not_le := Nat.lt_iff_le_and_not_ge
 
-protected theorem eq_or_lt_of_not_lt {a b : Nat} (hnlt : ¬ a < b) : a = b ∨ b < a :=
-  (Nat.lt_trichotomy a b).resolve_left hnlt
+protected theorem lt_iff_le_and_ne {a b : Nat} : a < b ↔ a ≤ b ∧ a ≠ b :=
+  ⟨fun h => ⟨Nat.le_of_lt h, Nat.ne_of_lt h⟩, fun h => Nat.lt_of_le_of_ne h.1 h.2⟩
+
+protected theorem le_antisymm_iff {a b : Nat} : a = b ↔ a ≤ b ∧ b ≤ a :=
+  ⟨fun | rfl => ⟨Nat.le_refl _, Nat.le_refl _⟩, fun ⟨hle, hge⟩ => Nat.le_antisymm hle hge⟩
+
+protected alias eq_iff_le_and_ge := Nat.le_antisymm_iff
 
-protected theorem not_lt_of_le {n m : Nat} (h₁ : m ≤ n) : ¬ n < m := (Nat.not_le_of_gt · h₁)
+protected theorem lt_connex {a b : Nat} : a ≠ b → a < b ∨ b < a := by
```

Comment:
This is a weird name. Is there any precedent?

Reply:
This is "connectedness" of a relation, but I agree it is not used in the lean naming convention. I would just call it `lt_or_gt_of_ne` (and ignore the part where it says `gt` even though it uses `lt`).

Reply:
Yes, there's a million synonyms in this section. So, in this case, the consensus is to just keep `lt_or_gt_of_ne` and delete both `lt_connex` and `lt_or_lt_of_ne`?